edgelet/iotedge/src/check/checks/proxy_settings.rs

```diff
@@ -0,0 +1,55 @@
+use failure::{self, Context};
+
+use crate::check::{Check, CheckResult, Checker, CheckerMeta};
+
+#[derive(Default, serde_derive::Serialize)]
+pub(crate) struct ProxySettings {}
+
+#[async_trait::async_trait]
+impl Checker for ProxySettings {
+    fn meta(&self) -> CheckerMeta {
+        CheckerMeta {
+            id: "proxy-settings",
+            description: "proxy settings are consistent in iotedged, moby daemon and config.toml",
+        }
+    }
+
+    async fn execute(&mut self, check: &mut Check) -> CheckResult {
+        self.inner_execute(check)
+            .await
+            .unwrap_or_else(CheckResult::Failed)
+    }
+}
+
+impl ProxySettings{
+    async fn inner_execute(&mut self, check: &mut Check) -> Result<CheckResult, failure::Error> {
+        let settings = if let Some(settings) = &mut check.settings {
+            settings
+        } else {
+            return Ok(CheckResult::Skipped);
+        };
+
+        // Pull the proxy address from the aziot-edged settings
+        // for Edge Agent's environment variables.
+        let edge_agent_proxy_uri = settings.base.agent.env().get("https_proxy");
+
+        // Pull the proxy address from the evironment for Moby and Edge Daemon
+        let moby_proxy_uri = std::env::var("HTTPS_PROXY")
+                                .ok();
+        let edge_daemon_proxy_uri = std::env::var("https_proxy")
```

Comment:
Similarly I think you should obtain the Edge daemon's proxy setting by checking the systemd config file for aziot-edged. It would be located in /etc/systemd/system/aziot-edged.service.d. Alternatively, it might be best to check the `Environment` properyt for the running aziot-edged daemon by running `sudo systemctl show --property=Environment aziot-edged` and parsing the output to check for the value of the https_proxy variable.